lib/jekyll/commands/contentful.rb

```diff
@@ -10,8 +10,11 @@ def self.init_with_program(prog)
 
           options.each {|opt| c.option(*opt) }
 
+          add_build_options(c)
+
           c.action do |args, options|
-            contentful_config = Jekyll.configuration['contentful']
+            jekyll_options = configuration_from_options(options)
```

Comment:
`configuration_from_options` is a method of the `Command` class, which simply wraps `Jekyll#configuration`. Since the plugin subclasses `Command`, I think it makes sense to use the inherited method rather then `Jekyll#configuration`.

It is necessary to pass in the `options` hash to make the `--config` flag, specifying a custom configuration file, work. If this flag is not properly supported and a user does not use the default `_config.yml` file, the plugin will crash because `jekyll_options['contentful']` (`Jekyll.configuration['contentful']` on the current master branch) will be `nil`.

The implementation of `Jekyll#configuration`, demonstrating how Jekyll defaults, the standard `_config.yml`, and any CLI options are parsed into the eventual configuration, can be found here: https://github.com/jekyll/jekyll/blob/master/lib/jekyll.rb#L93-L113